src/recipe.cpp

```diff
@@ -590,32 +590,73 @@ std::string recipe::required_proficiencies_string( const Character *c ) const
     return required;
 }
 
+std::string profstring( const std::pair<proficiency_id, std::pair<float, float>> &prof,
```

Comment:
I suggest defining a little three-member helper struct here rather than using nested `pair`s to represent the data.  Of just have three separate function arguments, and keep the helper struct within the `missing_proficiencies_string` function below.

Reply:
the nested pairs was definitely an error borne of being too tired to safely touch code.

i opted to define the struct outside, because there's two functions that would be calling `profstring`, and defining the struct twice seemed silly.